File: src/test/java/io/tripled/social/client/infrastructure/SystemDateTimeProviderTest.java
package io.tripled.social.client.infrastructure;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.CoreMatchers.notNullValue;
import static org.hamcrest.MatcherAssert.assertThat;

class SystemDateTimeProviderTest {

  private SystemDateTimeProvider dateTimeProvider;

  @BeforeEach
  void setUp() {
    dateTimeProvider = new SystemDateTimeProvider();
  }

  @Test
  void returnsNow() {
    assertThat(dateTimeProvider.now(), notNullValue());
  }
}

File: src/test/kotlin/io/tripled/social/client/infrastructure/SystemDateTimeProviderTest.kt
package io.tripled.social.client.infrastructure

import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.time.LocalDateTime

internal class SystemDateTimeProviderTest {
    private var dateTimeProvider: SystemDateTimeProvider? = null

    @BeforeEach
    fun setUp() {
        dateTimeProvider = SystemDateTimeProvider()
    }

    @Test
    fun returnsNow() {
        MatcherAssert.assertThat<LocalDateTime>(dateTimeProvider!!.now(), CoreMatchers.notNullValue())
    }
}